docs/src/solvers/nonlinear_system_solvers.md

```diff
@@ -0,0 +1,163 @@
+# [Nonlinear System Solvers](@id nonlinearsystemsolvers)
+
+```julia
+solve(prob::NonlinearProblem, alg; kwargs...)
+```
+
+Solves for ``f(u) = 0`` in the problem defined by `prob` using the algorithm `alg`. If no
+algorithm is given, a default algorithm will be chosen.
+
+## Recommended Methods
+
+The default method [`FastShortcutNonlinearPolyalg`](@ref) is a good choice for most
+problems. It is a polyalgorithm that attempts to use a fast algorithm ([`Klement`](@ref),
+[`Broyden`](@ref)) and if that fails it falls back to a more robust algorithm
+([`NewtonRaphson`](@ref)) before falling back the most robust variant of
+[`TrustRegion`](@ref). For basic problems this will be very fast, for harder problems it
+will make sure to work.
+
+If one is looking for more robustness then [`RobustMultiNewton`](@ref) is a good choice. It
+attempts a set of the most robust methods in succession and only fails if all of the methods
+fail to converge. Additionally, [`DynamicSS`](@ref) can be a good choice for high stability
+if the root corresponds to a stable equilibrium.
+
+As a balance, [`NewtonRaphson`](@ref) is a good choice for most problems that aren't too
+difficult yet need high performance, and  [`TrustRegion`](@ref) is a bit less performant but
+more stable. If the problem is well-conditioned, [`Klement`](@ref) or [`Broyden`](@ref) may
+be faster, but highly dependent on the eigenvalues of the Jacobian being sufficiently small.
+
+[`NewtonRaphson`](@ref) and [`TrustRegion`](@ref) are designed for for large systems. They
+can make use of sparsity patterns for sparse automatic differentiation and sparse linear
+solving of very large systems. Meanwhile, [`SimpleNewtonRaphson`](@ref) and
+[`SimpleTrustRegion`](@ref) are implementations which are specialized for small equations.
+They are non-allocating on static arrays and thus really well-optimized for small systems,
+thus usually outperforming the other methods when such types are used for `u0`.
```

Comment:
Mention GPUs as well? That should probably get a tutorial too. I think I have left a blank tutorial on static compilation for that kind of stuff.

Reply:
Mentioned it, I might do a section on GPU compilation for the paper, I will port that over when done.